### PR TITLE
CAMEL-13601 - Fix camel-jira client close and add logging

### PR DESCRIPTION
**Pull request by claudio4j**
https://issues.apache.org/jira/browse/CAMEL-13601

### CODE REVIEW

File: components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java [hunk 1]
```diff
@@ -43,15 +43,17 @@
 
     public NewCommentsConsumer(JiraEndpoint endpoint, Processor processor) {
         super(endpoint, processor);
-        LOG.info("JIRA NewCommentsConsumer: Indexing current issue comments...");
-        getComments(false);
+        LOG.info("JIRA NewCommentsConsumer: Indexing current issue comments.");
```
Comment (davsclaus) [inline components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java]: We should move such tasks to doStart and not in constructors

File: components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/AbstractJiraConsumer.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public AbstractJiraConsumer(JiraEndpoint endpoint, Processor processor) {
 
     // Ignore maxResults if it's <= 0.
     protected List<Issue> getIssues(String jql, int start, int maxPerQuery, int maxResults) {
-        LOG.info("Indexing current JIRA issues...");
+        LOG.info("Start indexing current JIRA issues...");
```
Comment (davsclaus) [inline components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/AbstractJiraConsumer.java]: Do we really need this INFO logging noise? If this method is invoked frequently you get spammed with logs. Can we lower this to DEBUG

File: components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java [hunk 3]
```diff
@@ -60,21 +62,25 @@ protected int poll() throws Exception {
     }
 
     // In the end, we want *new* comments oldest to newest.
-    private Stack<Comment> getComments(boolean loadComments) {
-        Stack<Comment> newComments = new Stack<>();
+    @SuppressWarnings("ConstantConditions")
+    private List<Comment> getComments() {
+        LOG.info("Start: Jira NewCommentsConsumer: retrieving issue comments. Last comment id: {}", lastCommentId);
```
Comment (davsclaus) [inline components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java]: Do we really need this INFO logging noise? If this method is invoked frequently you get spammed with logs. Can we lower this to DEBUG

File: components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java (hunk 1 above)
Comment (claudio4j) [reply components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/NewCommentsConsumer.java]: Done. Thanks for the review.

File: components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/AbstractJiraConsumer.java (hunk 2 above)
Comment (claudio4j) [reply components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/AbstractJiraConsumer.java]: Done. Thanks for the review.

File: parent/pom.xml [hunk 4]
```diff
@@ -384,7 +384,7 @@
         <jgroups-raft-jgroups-version>4.0.15.Final</jgroups-raft-jgroups-version>
         <jgroups-raft-leveldbjni-version>1.8</jgroups-raft-leveldbjni-version>
         <jgroups-raft-mapdb-version>1.0.8</jgroups-raft-mapdb-version>
-        <jira-guava-version>26.0-jre</jira-guava-version>
+        <jira-guava-version>20.0</jira-guava-version>
```
[the remark this answers is not in the document]
Comment (claudio4j) [reply parent/pom.xml]: The jira rest java client (JRJC) 5.1.0 doesn't define a guava version. As previously I used JRJC 5.1.2, its pom.xml defined the guava version, but when I changed to 5.1.0, I forgot to add the guava version.
https://bitbucket.org/atlassian/jira-rest-java-client/src/jira-rest-java-client-parent-5.1.0/pom.xml

**Comment by claudio4j** [comment]
Done, thanks for the review.

**Comment by oscerd** [comment]
What's the reason for Guava version downgrade?

File: parent/pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (oscerd) [reply parent/pom.xml]: ok. thanks

**Comment by oscerd** [comment]
Merged on master

**Comment by oscerd** [comment]
Thanks

**Pull request closed without merging**